streamingpro-core/src/main/java/tech/mlsql/crawler/RestUtils.scala

```diff
@@ -0,0 +1,342 @@
+    val requestInterval = JavaUtils.timeStringAsMs(config.getOrElse("config.request.interval", "10ms"))
+    val maxTries = config.getOrElse("config.retry", "3").toInt
+    val debug = config.getOrElse("config.debug", "false").toBoolean
+    val (status, content) = executeWithRetrying[(Int, String)](maxTries)((() => {
+      try {
+        val (status, content) = _http(url, method, params, headers, config)
+        Thread.sleep(requestInterval)
+        (status, content)
+      } catch {
+        case e: Exception =>
+          if (debug) {
+            val message = "request url: " + url + ", msg:" + ExceptionUtils.getMessage(e) + "\r\n\"stackTrace:" +
+              ExceptionUtils.getStackTrace(e)
+            logError(format(message))
+          }
+          (500, null)
+      }
+    }) (),
+      tempResp => {
+        val t = tempResp != null && tempResp._1 == 200
+        if (!t) {
+          Thread.sleep(retryInterval)
+        }
+        t
+      },
+      failResp => {}
+    )
+    (status, content)
+  }
+
+  def _http(url: String, method: String, params: Map[String, String],
+            headers: Map[String, String], config: Map[String, String]): (Int, String) = {
+    val httpMethod = new String(method).toLowerCase()
+    val context = ScriptSQLExec.contextGetOrForTest()
+    val debug = config.getOrElse("config.debug", "false").toBoolean
+    val request = httpMethod match {
+      case "get" =>
+
+        val finalUrl = if (params.nonEmpty) {
+          val urlParam = params.map { case (k, v) => s"$k=$v" }.mkString("&")
+          if (url.contains("?")) {
+            if (url.endsWith("?")) url + urlParam else url + "&" + urlParam
+          } else {
+            url + "?" + urlParam
+          }
+        } else url
+
+        Request.Get(finalUrl)
+
+      case "post" => Request.Post(url)
+      case "put" => Request.Put(url)
+      case v =>
+        if (debug) {
+          logError(format(s"request url: $url, Content-Type $v  is not support yet"))
+        }
+        return null
+    }
+
+    if (config.contains("socket-timeout")) {
+      request.socketTimeout(JavaUtils.timeStringAsMs(config("socket-timeout")).toInt)
+    }
+
+    if (config.contains("connect-timeout")) {
+      request.connectTimeout(JavaUtils.timeStringAsMs(config("connect-timeout")).toInt)
+    }
+
+    headers foreach { case (k, v) => request.setHeader(k, v) }
+    val contentTypeValue = headers.getOrElse("content-type", headers.getOrElse("Content-Type", "application/x-www-form-urlencoded"))
+    request.setHeader("Content-Type", contentTypeValue)
+
+    val response = (httpMethod, contentTypeValue) match {
+      case ("get", _) => request.execute()
+
+      case ("post", contentType) if contentType.trim.startsWith("application/json") =>
+        if (params.nonEmpty) {
+          try {
+            request.bodyString(JsonUtils.toJson(params), ContentType.APPLICATION_JSON).execute()
```

Comment:
It used to get json from params("body").

Modified to directly convert params to json to avoid misuse caused by directly passing parameters without passing the body.